.pre-commit-config.yaml

```diff
@@ -10,8 +10,10 @@ repos:
     rev: 'v0.4.3'
     hooks:
     - id: ruff
-      args: ["--fix", "--show-fixes"]
+      args: ["--fix", "--show-fixes", --select, I]
+      types_or: [ python, pyi, jupyter ]
     - id: ruff-format
+      types_or: [ python, pyi, jupyter ]
```

Comment:
These are the only config changes in this PR. Everything else was done by the linters.

Reply:
Noting that we should be able to pull the `--select I` bit out and put it in pyproject.toml

```
[RF102](https://learn.scientific-python.org/development/guides/style#RF102): isort must be selected
Must select the isort I checks. Recommended:
[tool.ruff.lint]
extend-select = [
  "I",  # isort
]

[RF103](https://learn.scientific-python.org/development/guides/style#RF103): pyupgrade must be selected
Must select the pyupgrade UP checks. Recommended:
[tool.ruff.lint]
extend-select = [
  "UP",  # pyupgrade
]
```